packages/wrangler/src/miniflare-cli/assets.ts

```diff
@@ -63,6 +68,64 @@ export default async function generateASSETSBinding(options: Options) {
 	};
 }
 
+/**
+ * An Undici custom Dispatcher that is used for the fetch requests
+ * of the Pages dev server proxy.
+ *
+ * Notably, the ProxyDispatcher reinstates the Host header that was removed by the
+ * Undici `fetch` function call. Undici removes the Host header as a security precaution,
+ * but this is not relevant for an internal Proxy.
+ *
+ * The ProxyDispatcher will delegate through to the current global `Dispatcher`,
+ * ensuring that the request is routed correctly in case a developer has changed the
+ * global Dispatcher by a call to `setGlobalDispatcher()`.
+ */
+class ProxyDispatcher extends Dispatcher {
+	private dispatcher: Dispatcher = getGlobalDispatcher();
+
+	constructor(private host: string | null) {
+		super();
+	}
+
+	dispatch(
+		options: Dispatcher.DispatchOptions,
+		handler: Dispatcher.DispatchHandlers
+	): boolean {
+		if (this.host !== null) {
+			ProxyDispatcher.reinstateHostHeader(options.headers, this.host);
+		}
+		return this.dispatcher.dispatch(options, handler);
+	}
+
+	close() {
+		return this.dispatcher.close();
+	}
+
+	destroy() {
+		return this.dispatcher.destroy();
+	}
+
+	/**
+	 * Ensure that the request contains a Host header, which would have been deleted
+	 * by the `fetch()` function before calling `dispatcher.dispatch()`.
+	 */
+	private static reinstateHostHeader(
+		headers: string[] | IncomingHttpHeaders | null | undefined,
+		host: string
+	) {
+		assert(headers, "Expected all proxy requests to contain headers.");
+		assert(
```

Comment:
Yeah sorry. I meant to write a comment about why I changed this but got dragged into a meeting.
I analysed the code in Undici and found that for this case it will always present a defined object.
So I removed the complexity of the assertions since they would never fail.

I realised that this is only valid as long as Undici doesn't change its behaviour. But I believe that this will only happen when we bump the version of Undici via a PR that would start to fail this assertion in our fixture tests.